### ISSUE TITLE
Run the docker image as non-root users

### ISSUE THREAD

**Issue by tschaffter** [data2health/nlp-sandbox-data-node-i2b2-2014#11]
By default, Docker containers run processes with an admin user. For security reason, it is recommended to make sure that a container uses a non-admin user.

Docker images:

- DB
- REST API

**Comment by thomasyu888** [comment]
https://docs.docker.com/engine/security/rootless/

**Comment by thomasyu888** [comment]
In case you don't want that, 

sudo usermod -aG docker $USER

Will allow your user to run docker commands without doing sudo.

**Comment by tschaffter** [comment]
Thanks @thomasyu888, this page is new to me. I've recently started to use `systemctl --user` for to run a few service like `pulseaudio`.

An easy change is to make sure that we ends the Dockerfile with current user being non-root. This is an easy change that do not requires the person deploying this data node to change anything to their local docker install. When WE deploy this data node in a production environment for the sake of benchmarking methods, we will follow the best practices that you have mentioned above. We will share this best practices in the README of this repo too.

**Comment by thomasyu888** [comment]
No problem @tschaffter.  Just to add: https://docs.docker.com/engine/install/linux-postinstall/#:~:text=The%20Docker%20daemon%20always%20runs,members%20of%20the%20docker%20group.. 

The Docker daemon will always run as the root user, this is the inherit risk of using Docker all together which is why technology like singularity exist: https://sylabs.io/guides/3.6/user-guide/.  I have not seen the convention of specifying the current user as being non-root in a Dockerfile.

**Comment by tschaffter** [comment]
This is achieved using `USER` in the dockerfile

>TL;DR

> One best practice when running a container is to launch the process with a non root user. This is usually done through the usage of the USER instruction in the Dockerfile. But, if this instruction is not present, it doesn’t necessarily mean the process is run as root.

[Source](https://medium.com/better-programming/running-a-container-with-a-non-root-user-e35830d1f42a)

**Comment by tschaffter** [comment]
There are cases where you can benefit from specifying the UserID and GroupID the container should use. 

Example: https://hub.docker.com/r/linuxserver/unifi-controller (see section "User / Group Identifiers")

**Comment by thomasyu888** [comment]
Ahh interesting.  I forget that there is a whole world of running docker containers as web services which I'm not super familiar with.  Thanks!

**Comment by tschaffter** [comment]
Done by dropping privileges using s6 before starting the api server, see #16
